PlutusV2 -> PlutusV3

locallycompact opened the pull request:
Updating the Hydra protocol transactions to use Plutus V3 validators.


---

<!-- Consider each and tick it off one way or the other -->
* [ ] CHANGELOG updated or not needed
* [ ] Documentation updated or not needed
* [ ] Haddocks updated or not needed
* [ ] No new TODOs introduced or explained herafter

locallycompact committed:
PlutusV2 -> PlutusV3

locallycompact commented:
This pull request should help debug this error

https://github.com/IntersectMBO/plutus/pull/6342

https://github.com/cardano-scaling/hydra/actions/runs/10078221725/job/27862634080

ch1bo commented on hydra-node/src/Hydra/Ledger/Cardano/Evaluate.hs:
```diff
@@ -448,184 +439,260 @@ genPointInTimeAfter deadline = do
 -- ** Plutus cost model fixtures
 
 -- | Current (2023-08-04) mainnet PlutusV2 cost model.
```
```suggestion
-- | Current (2024-08-13) preview PlutusV3 cost model.
```

ch1bo committed:
Use off-the-shelf ScriptContext in mHead validator

ch1bo committed:
Use PlutusScriptV3 everywhere

ch1bo committed:
Use PlutusV3 in more places

ch1bo committed:
FIXME: increase max tx size for local e2e tests

This should not be kept like this and is only incresased to progress
with V3 update

ch1bo committed:
Update local e2e test to use Conway protocol version 9

ch1bo committed:
Fix HeadTokens minting validator

The decoding of script context got lost when updating from PlutusV2 to
PlutusV3 seemingly.

locallycompact committed:
regenerate scripts

locallycompact committed:
Use off-the-shelf ScriptContext everywhere

locallycompact committed:
stash plutus-cbor

locallycompact committed:
change wrapMintingPolicy

locallycompact committed:
Regenerate scripts

locallycompact committed:
Bump maxTXSize and executionMemory

locallycompact commented:
Upgrading to PlutusV3 works functionally in terms of logic but exceed Tx size constraints and memory units, especially using off-the-shelf ScriptContext. Restoring custom ScriptContext or using techniques like https://github.com/cardanoapi/hardfork-testing/blob/main/migration.md will be needed to optimize performance.

locallycompact committed:
Bump executionMemory

locallycompact committed:
Bump maxTxSize

locallycompact committed:
Restore old script context for benchmarks

locallycompact committed:
treefmt

locallycompact committed:
hlint

ch1bo commented:
@locallycompact Okay. Let's summarize the findings in the monthly report and park this effort for now.
